tests/unit/onyxTest.js

```diff
@@ -306,4 +306,80 @@ describe('Onyx', () => {
                 });
             });
     });
+
+    it('should use update data object to set/merge keys', () => {
+        let testKeyValue;
+        connectionID = Onyx.connect({
+            key: ONYX_KEYS.TEST_KEY,
+            initWithStoredValues: false,
+            callback: (value) => {
+                testKeyValue = value;
+            },
+        });
+
+        let anotherTestKeyValue;
+        connectionID = Onyx.connect({
+            key: ONYX_KEYS.ANOTHER_TEST,
+            initWithStoredValues: false,
+            callback: (value) => {
+                anotherTestKeyValue = value;
+            },
+        });
+
+        return waitForPromisesToResolve()
+            .then(() => {
+                Onyx.set(ONYX_KEYS.TEST_KEY, true);
+                Onyx.set(ONYX_KEYS.ANOTHER_TEST, {test1: 'test1'});
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                expect(testKeyValue).toBe(true);
+                expect(anotherTestKeyValue).toEqual({test1: 'test1'});
+                Onyx.update([
+                    {
+                        onyxMethod: 'set',
+                        key: ONYX_KEYS.TEST_KEY,
+                        value: 'four',
+                    },
+                    {
+                        onyxMethod: 'merge',
+                        key: ONYX_KEYS.ANOTHER_TEST,
+                        value: {test2: 'test2'},
+                    },
+                ]);
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                expect(testKeyValue).toBe('four');
+                expect(anotherTestKeyValue).toEqual({test1: 'test1', test2: 'test2'});
+            });
+    });
+
+    it('should throw an error when the data object is incorrect in Onyx.update', () => {
+        const data = [
+            {onyxMethod: 'set', key: ONYX_KEYS.TEST_KEY, value: 'four'},
+            {onyxMethod: 'multiSet', key: ONYX_KEYS.ANOTHER_TEST, value: {test2: 'test2'}}
+        ];
+
+        try {
+            Onyx.update(data);
+        } catch (error) {
+            expect(error.message).toEqual('Invalid onyxMethod multiSet in Onyx update.');
+        }
+
+        try {
+            data[1] = {onyxMethod: 'merge', key: true, value: {test2: 'test2'}};
+            Onyx.update(data);
+        } catch (error) {
+            // eslint-disable-next-line max-len
```

Comment:
Is this necessary? It looks like another liner higher up is longer than this, and I thought the length was something like 190, which this line looks to be an OK length.

Reply:
Yea, I agree with not having warnings. I guess I wonder why it's at 120 and not 190. Maybe try updating the eslint config to make sure it's using our latest?

Reply:
Yea, we are really outdated on eslint-config. Updating it surfaces A LOT of errors, so I'll address that in a separate PR. I'll keep the comment for now then.

Reply:
Got a [PR up](https://github.com/Expensify/react-native-onyx/pull/135) to update eslint-config